discovery/sync_manager.go

```diff
@@ -0,0 +1,673 @@
+package discovery
+
+import (
+	"container/list"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/lightningnetwork/lnd/lnpeer"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/routing"
+	"github.com/lightningnetwork/lnd/ticker"
+)
+
+const (
+	// DefaultSyncerRotationInterval is the default interval in which we'll
+	// rotate a single active syncer.
+	DefaultSyncerRotationInterval = 20 * time.Minute
+
+	// DefaultHistoricalSyncInterval is the default interval in which we'll
+	// force a historical sync to ensure we have as much of the public
+	// network as possible.
+	DefaultHistoricalSyncInterval = 20 * time.Minute
```

Comment:
an idea to avoid the rotation and historical ticker ticking at the same time?

Reply:
Discussed offline, most likely won't be an issue.